### ISSUE TITLE
add_type always fails

### ISSUE THREAD

**Issue by TheMaddScientist** [h2non/filetype.py#70]
If you attempt to do an add_type with a subclass of Type, you always get the "instance must inherit from filetype.types.Type". This appears to be because isinstance only returns true for actual instances and not for subclasses. You need to use issubclass to check for subclasses. I am using Python 3.8, so this may be new behavior.

If I fix this error, I get a further buffer error.  Attached is a zip file with my example code. You will need to change the file locations.
[detect_file_type.zip](https://github.com/h2non/filetype.py/files/4871101/detect_file_type.zip)

**Comment by dosas** [comment]
@TheMaddScientist No, ```isintance``` is the correct call but you need to add an instance of your ```tLilyPond``` class!
```
ft.add_type(tLilyPond())
```
